bitcoin/src/bip32.rs

```diff
@@ -733,26 +774,66 @@ impl Xpub {
     /// Attempts to derive an extended public key from a path.
     ///
     /// The `path` argument can be any type implementing `AsRef<ChildNumber>`, such as `DerivationPath`, for instance.
-    pub fn derive_pub<C: secp256k1::Verification, P: AsRef<[ChildNumber]>>(
+    pub fn derive_pub<C: secp256k1::Verification>(
         &self,
         secp: &Secp256k1<C>,
-        path: &P,
+        path: NormalDerivationPath,
+    ) -> Xpub {
+        let mut pk: Xpub = *self;
+        for normal_cnum in path.into_iter() {
+            pk = pk
+                .internal_derive_pub(secp, ChildNumber::Normal(normal_cnum.0))
+                .expect("Unreachable: NormalChildNumber is always valid");
```

Comment:
But that's not the only reason it could fail:
https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#L96

Reply:
The other reason is cryptographically impossible, isn't it?

Reply:
Not impossible, but highly unlikely.
I don't think we should worry about that, since the current implementation (the one that is standing now and that the PR is changing) also does not deal with that.
I think it would be handled by `secp256k1` but I'm not certain, please correct me if I'm wrong.

Reply:
You're more likely to get the same address as some other person has. That's enough to rule out needing to care about it.

Reply:
"cryptographically impossible" means impossible. If we are going to consider 2^-128 events then there is no point even reviewing the code because it's far more likely that you have been drugged into hallucinating this entire project.

Reply:
offtopic:
Come on guys, I come from a hardcore stats and probability background. Hence, impossible is $P=0$.
"Cryptographically impossible" means [negligible probability](https://en.wikipedia.org/wiki/Negligible_function)?

Reply:
Sure, but that definition of "impossible" has no real-world referent. One might say it's ... impossible ;).

(I'll stop here. I think we all understand each other. Though I guess, if somebody wants to update CONTRIBUTING.md to say that our official policy for 2^-100 events is to panic on them, I'm fine with that.)

Reply:
So, resolved ?